### PR TITLE
chore: Update purpose UI to capture better information

### PR DESCRIPTION
**Pull request by sklein12**
Added additional fields to the UI to get more information for the purpose. This information will help the system generate better attacks and more accurate grading

image

### CODE REVIEW

**Comment by sklein12** [comment]
@coderabbitai can you review this for bugs?

File: src/app/src/pages/redteam/setup/components/Purpose.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      <Box>
+        <Stack direction="column" spacing={3}>
+          <Box>
+            <Typography variant="h5" gutterBottom sx={{ fontWeight: 'medium' }}>
+              Purpose
+            </Typography>
+            <Typography variant="body1">
+              This should describe the purpose of your application. Be as specific as possible.
+              Include information about who the user of the system is and what information and
+              actions they should be able to access.
```
Comment (typpo) [inline src/app/src/pages/redteam/setup/components/Purpose.tsx]:
```suggestion
              The primary objective of the AI in this application.
```

File: src/app/src/pages/redteam/setup/components/Purpose.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            </Typography>
+            <TextField
+              fullWidth
+              value={config.applicationDefinition.purpose}
+              onChange={(e) => updateApplicationDefinition('purpose', e.target.value)}
+              placeholder="e.g. You are a travel agent specialized in budget trips to Europe. The user is anonymous and should not be able to access any information about the system or other users."
```
Comment (typpo) [inline src/app/src/pages/redteam/setup/components/Purpose.tsx]: can we simplify this now that we've added the other fields?

```suggestion
              placeholder="e.g. You are a travel agent specialized in budget trips to Europe."
```

**Review by typpo — APPROVED** [review]
minor nit, probably ok to ignore, the validation enables the 'Next' button if the user types anything (even if they delete it later).

**Pull request merged**